src/Features/Core/Portable/Completion/CompletionService.ProviderManager.cs

```diff
@@ -93,17 +93,22 @@ private ValueTask ProcessBatchAsync(ImmutableSegmentedList<IReadOnlyList<Analyze
                 return ValueTaskFactory.CompletedTask;
             }
 
-            public ImmutableArray<CompletionProvider> GetCachedProjectCompletionProvidersOrQueueLoadInBackground(Project? project)
+            public ImmutableArray<CompletionProvider> GetCachedProjectCompletionProvidersOrQueueLoadInBackground(Project? project, CompletionOptions options)
             {
                 if (project is null || project.Solution.WorkspaceKind == WorkspaceKind.Interactive)
                 {
                     // TODO (https://github.com/dotnet/roslyn/issues/4932): Don't restrict completions in Interactive
                     return ImmutableArray<CompletionProvider>.Empty;
                 }
 
-                // Don't load providers if they are not already cached,
-                // return immediately and load them in background instead.
+                // On primary completion paths, don't load providers if they are not already cached,
+                // return immediately and load them in background instead. If the test hook
+                // 'ForceExpandedCompletionIndexCreation' is set, calculate the values immediately.
                 // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1620947
+                if (options.ForceExpandedCompletionIndexCreation)
```

Comment:
The fix looks good to me. But I'd suggest to change the name of this option to be more general to reflect this change.

Reply:
➡️ I left the option name the same, but updated the comment